File: Book/C/Fig_13.1_taskBug.c
// sample compile command: "gcc -fopenmp -c Fig_13.1_taskBug.c" to generate *.o object file

void work()
{
   #pragma omp task  //task 1
   {         
   
      #pragma omp task   //task 2
      {   
      
              #pragma omp critical // Critical region 1
               {/* do work here */}
      }
      #pragma omp critical // Critical Region 2
      {
         
         #pragma omp task // task 3
            {/* do work here */}  
      }
   }
}

File: Book/Fortran/Fig_13.1_taskbug.f90
! sample compile command to generate *.o object file:
!      gfortranc -fopenmp -c Fig_13.1_taskBug.f90

subroutine  work()
   use omp_lib
   implicit none

   !$omp task   ! task 1
      !$omp task   ! task 2
          !$omp critical    ! Critical region 1
             ! do work here
          !$omp end critical  ! end Critical Region 1
      !$omp end task  ! end task2
      !$omp critical  ! Critical Region 2
         !$omp task  ! task 3
             ! do work here } 
         !$omp end task  ! end task3
      !$omp end critical ! end Critical Region 2
   !$omp end task    ! end task 1
end subroutine 
